core/src/main/java/eu/fasten/core/plugins/GraphDBConnector.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.fasten.core.plugins;
+
+import org.rocksdb.RocksDBException;
+
+/**
+ * A plug-in that needs to access graph database (RocksDB)
+ */
+public interface GraphDBConnector extends FastenPlugin {
```

Comment:
This interface does not look very useful. A graph connector should return a reference to an initialised `GraphDBDAO` to plug-ins so that the plug-in can directly query, same as the PostgresConnector.  A plug-in should not care about the location of the underlying database.

Reply:
I can change it to `setRocksDao(RocksDao rocksDao)` and then we can either create RocksDao instance in Fasten Server when initializing plugin or have a separate class RocksDBConnector with one static method that will take directory and return RocksDao instance (quite similar to PostgresConnector). Which one should I implement?

Reply:
The later. Is RocksDB protecting itself (e.g. at the filesystem level) against multiple writers? If not, we need to design for that.

Reply:
Hm, now that I think of it I am not sure that `setRocksDao(...)` would be appropriate. Since RocksDao is located in graph plugin module, we would have to have a dependency to graph plugin from core or server. I am not sure If that's a good solution.

Reply:
In order to solve the issue with circular dependency, I could move `RocksDao` to core. Do you think that would be appropriate @gousiosg ?

Reply:
Yes